src/arc_spice/variational_pipelines/utils.py

```diff
@@ -0,0 +1,489 @@
+import logging
+from functools import partial
+from typing import Any
+
+import torch
+from torch.nn.functional import softmax
+from transformers import AutoModelForSequenceClassification, AutoTokenizer, Pipeline
+
+logger = logging.Logger("RTC_variational_pipeline")
+
+
+class DummyPipeline:
```

Comment:
what's the purpose of this class if it will raise NotImplementedError when it's called?

Reply:
This is for tests so that dummy pipelines can be loaded without the overhead of the models themselves. I also use this class when initialising the pipelines of the base class, this was to ensure mypy didn't raise an attribute error when referencing the pipelines within the class.

Reply:
if it's only for use in tests then can it live in the test code?

Reply:
Sorry, this is still being used in the base class at initialisation. I've changed the error to AttributeError, as this will reflect the nature of the issue if it is called without being properly initialised?